Code/GraphMol/RGroupDecomposition/RGroupData.cpp

```diff
@@ -94,26 +94,16 @@ std::string RGroupData::toString() const {
 }
 
 void RGroupData::computeIsHydrogen() {  // is the rgroup all Hs
-  for (const auto &mol : mols) {
-    if (!isMolHydrogen(*mol)) {
-      is_hydrogen = false;
-      return;
-    }
-  }
-  is_hydrogen = true;
+  is_hydrogen = std::all_of(mols.begin(), mols.end(), [this](const auto &mol) {
+    return isMolHydrogen(*mol);
+  });
 }
 
-bool RGroupData::isMolHydrogen(ROMol &mol) {
-  for (ROMol::AtomIterator atIt = mol.beginAtoms(); atIt != mol.endAtoms();
-       ++atIt) {
-    auto atom = *atIt;
-    if (atom->getAtomicNum() > 1) {
-      return false;
-    } else if (atom->getAtomicNum() == 0 && !atom->hasProp(SIDECHAIN_RLABELS)) {
-      return false;
-    }
-  }
-  return true;
+bool RGroupData::isMolHydrogen(const ROMol &mol) const {
+  auto atoms = mol.atoms();
+  return std::all_of(atoms.begin(), atoms.end(), [](const auto &atom) {
```

Comment:
Interestingly, I find this more clear than the previous...

Reply:
You see? You are already hooked :-)